Manual review documentation (#769)
vinyldns/vinyldns: modules/docs/src/main/tut/api/batchchange-model.md

-   The ability to accept multiple changes in a single API call.
-   The ability to include records of multiple record types across multiple zones.
-   Input names are entered as fully-qualified domain names (or IP addresses for **PTR** records), so users don't have to think in record/zone context.
-   All record validations are processed simultaneously. [Fatal errors](../api/batchchange-errors/#fatal-errors) for any
change in the batch will result in a **400** response and none will be applied.

> -   Support for manual review if configured on your VinylDNS instance where batches that contain only [non-fatal errors](../api/batchchange-errors/#non-fatal-errors)

Reviewer: This was hard to follow. 
```suggestion
-   Support for [manual review](idk where to link to though) if enabled in your VinylDNS instance.
```

enter a manual review state. Batch change will remain in limbo until a system administrator (ie. support or super user)
either rejects it resulting in an immediate failure or approves it resulting in revalidation and submission for processing.

> -   Support for [notifications](../operator/config-api#additional-configuration-settings) when a batch change is rejected or implemented.

Reviewer 2: should we add "Support for scheduled changes by setting a scheduled time field if scheduled changes are configured for your VinylDNS instance"?

-   The ability to accept multiple changes in a single API call.
-   The ability to include records of multiple record types across multiple zones.
-   Input names are entered as fully-qualified domain names (or IP addresses for **PTR** records), so users don't have to think in record/zone context.
-   All record validations are processed simultaneously. [Hard errors](../api/batchchange-errors/#hard-errors) for any
change in the batch will result in a **400** response and none will be applied.
-   Support for manual review if configured on your VinylDNS instance where batches that contain only [soft errors](../api/batchchange-errors/#soft-errors)
enter a manual review state. Batch change will remain in limbo until a system administrator (ie. support or super user)
either rejects it resulting in an immediate failure or approves it resulting in revalidation and submission for processing.
-   Support for [notifications](../../operator/config-api#additional-configuration-settings) when a batch change is rejected or implemented.

> -   Support for [scheduled changes](../../operator/config-api#additional-configuration-settings) by setting a scheduled time field if scheduled changes are configured for your VinylDNS instance.

Reviewer 3: hmm, does this section exist? i feel like we have some mentions of scheduled throughout but didnt really cover the topic, and definitely have things missing. im inclined to just remove and and have a separate docs update with the scheduled stuff

Author: Removed; think we are planning to add scheduled documentation separately.

-   The ability to accept multiple changes in a single API call.
-   The ability to include records of multiple record types across multiple zones.
-   Input names are entered as fully-qualified domain names (or IP addresses for **PTR** records), so users don't have to think in record/zone context.
-   All record validations are processed simultaneously. [Fatal errors](../api/batchchange-errors/#fatal-errors) for any change in the batch will result in a **400** response and none will be applied.

> -   Support for manual review where batches that contain only [permissible errors](../api/batchchange-errors/#permissible-errors) enter a pending review state. Batch change will remain in limbo until a system administrator (ie. support or super user) either rejects it resulting in an immediate failure or approves it resulting in revalidation and submission for processing.

Reviewer 3: add an "if configured on for your vinyldns instance" somewhere (or something to that effect)

userId        | string      | The unique identifier of the user that created the batch change. |
userName      | string      | The username of the user that created the batch change. |
comments      | string      | Optional comments about the batch change. |
createdTimestamp | date-time      | The timestamp (UTC) when the batch change was created. |
changes       | Array of SingleChange | Array of single changes within a batch change. A *SingleChange* can either be a [SingleAddChange](#singleaddchange-attributes) or a [SingleDeleteChange](#singledeletechange-attributes). |

> status        | BatchChangeStatus | **PendingProcessing** - at least one change in batch in still in pending processing (will be or is currently auto-processing) state; **Complete** - all changes are in complete state; **Failed** - all changes are in failure state; **PartialFailure** - some changes have failed and the rest are complete; **PendingReview** - one or more changes requires manual [approval](../api/approve-batchchange)/[rejection](../api/reject-batchchange) by a system administrator (ie. super or support user) to proceed ; **Rejected** - the batch change was rejected by a system administrator (ie. super or support user) and no changes were applied; **Scheduled** - the batch change is scheduled for a later date at which it needs to be approved to proceed. |

Reviewer: ```suggestion
status        | BatchChangeStatus | **PendingProcessing** - at least one change has not finished processing; **Complete** - all changes have been processed successfully; **Failed** - all changes failed during processing; **PartialFailure** - some changes have failed and the rest were successful; **PendingReview** - one or more changes requires manual [approval](../api/approve-batchchange)/[rejection](../api/reject-batchchange) by a system administrator (ie. super or support user) to proceed ; **Rejected** - the batch change was rejected by a system administrator (ie. super or support user) and no changes were applied; **Scheduled** - the batch change is scheduled for a later date at which time it needs to be approved to proceed through processing. |
```

userId        | string      | The unique identifier of the user that created the batch change. |
userName      | string      | The username of the user that created the batch change. |
comments      | string      | Optional comments about the batch change. |
createdTimestamp | date-time      | The timestamp (UTC) when the batch change was created. |
changes       | Array of SingleChange | Array of single changes within a batch change. A *SingleChange* can either be a [SingleAddChange](#singleaddchange-attributes) or a [SingleDeleteChange](#singledeletechange-attributes). |

> status        | BatchChangeStatus | **PendingProcessing** - at least one change in batch in still in pending processing state; **Complete** - all changes are in complete state; **Failed** - all changes are in failure state; **PartialFailure** - some changes have failed and the rest are complete; **PendingReview** - one or more changes requires manual [approval](../api/approve-batchchange)/[rejection](../api/reject-batchchange) by a system administrator (ie. super or support user) to proceed ; **Rejected** - the batch change was rejected by a system administrator (ie. super or support user) and no changes were applied; **Scheduled** - the batch change is scheduled for a later date at which it needs to be approved to proceed. |

Reviewer 3: again here, i think it's not clear that PendingProcessing means the batch will be auto processed (or is in the the progress of being processed). With PendingReview in the mix, i just want to make it clear this thing isnt waiting on any intervention

type          | RecordType    | Type of DNS record, supported records for batch changes are currently: **A**, **AAAA**, **CNAME**, and **PTR**. |
ttl           | long          | The time-to-live in seconds. |
record        | [RecordData](../api/recordset-model/#record-data)    | The data added for this record, which varies by record type. | 
status        | SingleChangeStatus | Status for this change. Can be one of: **Pending**, **Complete**, **Failed**, **NeedsReview** or **Rejected**. |
recordName    | string        | The name of the record. Record names for the apex will be match the zone name (including terminating dot). |
zoneName      | string        | The name of the zone. |
zoneId        | string        | The unique identifier for the zone. |
systemMessage | string        | Conditional: Returns system message relevant to corresponding batch change input. |
recordChangeId| string        | Conditional: The unique identifier for the record change; only returned on successful batch creation. |
recordSetId   | string        | Conditional: The unique identifier for the record set; only returned on successful batch creation, |

> validationErrors | Array of BatchChangeError | Array containing any validation errors associated with this SingleAddChange. |

Reviewer 3: note that these will only exist on NeedsReview (or Rejected) single changes

changeType    | ChangeInputType | Type of change input. Can either be an **Add** or **DeleteRecordSet**. |
inputName     | string        | The fully-qualified domain name of the record which was provided in the create batch request. |
type          | RecordType    | Type of DNS record, supported records for batch changes are currently: **A**, **AAAA**, **CNAME**, and **PTR**. | 
status        | SingleChangeStatus | Status for this change. Can be one of: **Pending**, **Complete**, **Failed**, **NeedsReview** or **Rejected**. |
recordName    | string        | The name of the record. Record names for the apex will be match the zone name (including terminating dot). |
zoneName      | string        | The name of the zone. |
zoneId        | string        | The unique identifier for the zone. |
systemMessage | string        | Conditional: Returns system message relevant to corresponding batch change input. |
recordChangeId| string        | Conditional: The unique identifier for the record change; only returned on successful batch creation. |
recordSetId   | string        | Conditional: The unique identifier for the record set; only returned on successful batch creation, |

> validationErrors | Array of BatchChangeError | Array containing any validation errors associated with this SingleDeleteChange. |

Reviewer 3: ditto https://github.com/vinyldns/vinyldns/pull/769/files#r307828167
